code/game/objects/items/mop.dm

```diff
@@ -62,13 +60,10 @@
 		return
 
 	if(T)
-		user.visible_message(span_notice("[user] begins to clean \the [T] with [src]."), span_notice("You begin to clean \the [T] with [src]..."))
-		var/clean_speedies = 1
-		if(user.mind)
-			clean_speedies = user.mind.get_skill_modifier(/datum/skill/cleaning, SKILL_SPEED_MODIFIER)
-		if(do_after(user, mopspeed*clean_speedies, target = T))
-			to_chat(user, span_notice("You finish mopping."))
-			clean(T, user)
+		if(reagents.has_chemical_flag(REAGENT_CLEANS, 1))
+			SEND_SIGNAL(src, COMSIG_START_CLEANING, T, user)
+		else //won't clean the turf nor give you experience without cleaning reagents
+			SEND_SIGNAL(src, COMSIG_START_CLEANING, T, user, FALSE)
```

Comment:
instead of sending 2 signals, and since has_chemical_flag() only returns TRUE/FALSE, can't we do something instead like
```suggestion
		var/should_clean = reagents.has_chemical_flag(REAGENT_CLEANS, 1)
		SEND_SIGNAL(src, COMSIG_START_CLEANING, T, user, clean_target = should_clean)
```